<launch>
    <!-- 跟踪节点 -->
    <node name="xiaoqiang_tracking_node" pkg="xiaoqiang_track" type="tracking_node" output="screen">
        <!-- 人体识别程序选项，目前支持baidu和body_pose -->
        <param name="pose_tracker" value="baidu" />
        <!-- baidu人体识别app_id,此参数要去百度开发者中心自己申请 -->
        <param name="app_id" value="11468688" type="string"/>
        <!-- baidu人体识别api_key,此参数要去百度开发者中心自己申请 -->
        <param name="api_key" value="zYDLPERMvzaPpjlChkO6ZO4K"/>
        <!-- baidu人体识别secret_key,此参数要去百度开发者中心自己申请 -->
        <param name="secret_key" value="D6xBAQwgLnVGIv0sntSRDCG5awM3NdrA" />
        <!-- 超时时间，超过这个时间则认为调用失败 -->
        <param name="timeout" value="5000" />
        <!-- 主跟踪程序算法 -->
        <param name="tracker_main" value="MOSSE" />
        <!-- 辅助跟踪算法，当主跟踪算法失效时使用辅助算法的结果 -->
        <param name="tracker_aided" value="MEDIANFLOW" />
        <!-- 跟踪图像的话题 -->
        <remap from="~image" to="/camera_node/image_raw" />
        <!-- 语音功能话题 -->
        <remap from="~text" to="/xiaoqiang_tts/text" />
    </node>

    <!-- 跟踪运动控制节点 -->
    <node name="tracking_controller_node" pkg="xiaoqiang_track" type="controller_node.py">
        <!-- 运动控制PID参数中的P -->
        <param name="p" value="0.007" />
        <!-- 运动控制PID参数中的I -->
        <param name="i" value="0.0045" />
        <!-- 运动控制PID参数中的D -->
        <param name="d" value="0.001" />
        <!-- 运动控制PID的采样率 -->
        <param name="sample_rate" value="30" />
        <!-- 最大角速度，单位为 rad/s -->
        <param name="max_angle_speed" value="1" />
        <!-- 最大线速度，单位为 m/s -->
        <param name="max_linear_speed" value="0.4" />
        <!-- 底盘速度控制topic -->
        <remap from="~cmd_vel" to="/cmd_vel" />
        <!-- 跟踪的目标位置topic -->
        <remap from="~target" to="/xiaoqiang_tracking_node/target" />
    </node>

    <!-- 语音合成节点 -->
    <include file="$(find xiaoqiang_tts)/launch/tts_xunfei.launch" />
    <!-- 摄像头节点，如果不是小强，要修改成自己的摄像头节点 -->
    <include file="$(find usb_cam)/launch/ov2610.launch" />
</launch>